akka-actor-typed-tests/src/test/scala/akka/actor/typed/SupervisionSpec.scala

```diff
@@ -571,6 +571,44 @@ class SupervisionSpec extends ScalaTestWithActorTestKit(
       childProbe.expectNoMessage()
     }
 
+    "update failed behavior if exception from unstashAll" in {
+      testPreRestartSignalOnExceptionFromUnstashAll(strategy = SupervisorStrategy.restart)
+    }
+
+    def testPreRestartSignalOnExceptionFromUnstashAll(strategy: SupervisorStrategy): Unit = {
+
```

Comment:
This test is very similar to other unstash tests, need to be more sure how to verify the updated failed behavior.

Reply:
I wrote this, more concrete, test case to understand the problem better:

```scala
"signal PreRestart to the latest unstashed behavior" in {
  val probe = TestProbe[Any]()
  val stashingBehavior =
    Behaviors.setup[String] { ctx ⇒
      val stash = StashBuffer[String](10)
      def unstashing(n: Int): Behavior[String] =
        Behaviors.receiveMessage[String] {
          case "stash" ⇒
            probe.ref ! s"unstashing-$n"
            unstashing(n + 1)
          case "stash-fail" ⇒
            throw new TestException("unstash-fail")
        }.receiveSignal {
          case (_, PreRestart) ⇒
            probe.ref ! s"pre restart in state $n"
            Behaviors.same
        }

      Behaviors.receiveMessage[String] {
        case msg if msg.startsWith("stash") ⇒
          stash.stash(msg)
          Behavior.same
        case "unstash" ⇒
          stash.unstashAll(ctx, unstashing(0))
      }
    }
  val ref =
    spawn(Behaviors.supervise(stashingBehavior)
      .onFailure[TestException](SupervisorStrategy.restart))

  ref ! "stash"
  ref ! "stash-fail"
  ref ! "unstash"
  probe.expectMessage("unstashing-0")
  probe.expectMessage("pre restart in state 0")
}
```

Reply:
@johanandren this is great thanks! It much more clearly written for the problem too.

Reply:
I think it should maybe be verifying "pre start in state 1" though, as the first unstash succeeds, and the second fails, the second unstash iteration should get the PreRestart